modules/cbMap/processmap/importCSV.php

```diff
@@ -0,0 +1,175 @@
+    $params = array();
+    $index_query = $adb->pquery($index_q, $params);
+    $nr_rows = $adb->num_rows($index_query);
+    if ($nr_rows>0) {
+        $allids = array();
+        if ($options['update'] == 'FIRST') {
+            $allids[] = $adb->query_result($index_query, 0, $focus->table_index);
+        } elseif ($options['update'] == 'LAST') {
+            $allids[] = $adb->query_result($index_query, $nr_rows - 1, $focus->table_index);
+        }
+        if ($options['update'] == 'ALL') {
+            for ($i = 0; $i < $nr_rows; $i++) {
+                $allids[] = $adb->query_result($index_query, $i, $focus->table_index);
+            }
+        } 
+        for ($el = 0; $el < count($allids); $el++) {
+            $index_result = $adb->query_result($index_query, $el, $focus->table_index);
+              if($nr_rows>0)  $focus->retrieve_entity_info($index_result, $module);
+                 foreach ($updateFld as $upkey => $upVal) {
+                    $predefined = $upVal['predefined'];
+                    $value = $upVal['value'];
+                    if ($predefined == 'AUTONUM')
+                        $focus->column_fields[$upkey] = $el;
+                    else if(isset($upVal['relatedFields']) && !empty($upVal['relatedFields'])) {
+                            $relInformation = $upVal['relatedFields'][0];
+                            $relModule = $relInformation['relmodule'];
+                            $linkField = $relInformation['linkfield'];
+                            $fieldName = $relInformation['fieldname'];
+                            $otherid = $data[$linkField];
+
+                            if (!empty($otherid)) {
+                                include_once "modules/$relModule/$relModule.php";
+                                $otherModule = CRMEntity::getInstance($relModule);
+                                $customfld1 = $otherModule->customFieldTable;
+                                $index_rel = $adb->query("SELECT $otherModule->table_name.$otherModule->table_index
+                                FROM $otherModule->table_name
+                                INNER JOIN vtiger_crmentity ON vtiger_crmentity.crmid=$otherModule->table_name.$otherModule->table_index
+                                INNER JOIN $customfld1[0] ON $customfld1[0].$customfld1[1]=$otherModule->table_name.$otherModule->table_index
+                                WHERE vtiger_crmentity.deleted=0 and $fieldName='$otherid'");
+                                $focus->column_fields[$upkey] =$adb->query_result($index_rel,0);
+                                             
+                        }
+                    }
+                    elseif (!empty($data[$value]))
+                        $focus->column_fields[$upkey] = $data[$value];
+                    else
+                        $focus->column_fields[$upkey] = $predefined;
+                }
+                $focus->mode = 'edit';
+                $focus->id = $index_result;
+
```

Comment:
don't you have to sanitize data here? are special characters or non-iso dates and currency working correctly?